### PR TITLE
use Yarn instead of NPM

### PR DESCRIPTION
**Pull request by akrantz**
@TCourtneyOwen This modifies the Lerna configuration to use Yarn instead of NPM.

This has the effect of hoisting the packages to the root node_modules folder. I'm not sure whether this is desirable or not, but wanted to experiment. Yarn is faster and the Yarn Workspaces provides the hoisting/symlinking of packages in the package manager which others seem to use. I'm not sure what the downsides are -- if any.

### CODE REVIEW

File: packages/custom-functions-metadata/package.json [hunk 1]
```diff
@@ -32,7 +32,6 @@
   },
   "devDependencies": {
     "@types/assert": "^1.4.2",
-    "@types/commander": "^2.12.2",
```
Comment (TCourtneyOwen) [inline packages/custom-functions-metadata/package.json]: should types/commander in fact be removed here?

File: packages/office-addin-lint/package.json [hunk 2]
```diff
@@ -33,7 +33,6 @@
   },
   "devDependencies": {
     "@types/assert": "^1.4.2",
-    "@types/commander": "^2.12.2",
```
Comment (TCourtneyOwen) [inline packages/office-addin-lint/package.json]: same question here about whether this is intentionally removed

File: packages/custom-functions-metadata/package.json (hunk 1 above)
Comment (akrantz) [reply packages/custom-functions-metadata/package.json]: Yes, yarn gives a warning that "commander" includes types, so "@types/commander" should not be used.

File: packages/office-addin-lint/package.json (hunk 2 above)
Comment (akrantz) [reply packages/office-addin-lint/package.json]: same answer.  :-)

**Comment by TCourtneyOwen** [comment]
Looks like install fails in the CI loop.  Any idea what the problem is?

**Comment by akrantz** [comment]
I'm not planning to merge this right now. That's why it is a draft PR.

**Comment by akrantz** [comment]
Yes, you need to change the loop to use Yarn instead of NPM, e.g "yarn install", "yarn run build", etc.

This is one reason I think it would be good to move to yaml files because you can make PRs that include build pipeline changes as part of the PR.

**Pull request closed without merging**